pkg/sethandler/controller.go

```diff
@@ -233,3 +246,50 @@ func (setHandler *SetHandler) applyCpusetToContainer(containerID string, cpuset
 	}
 	return nil
 }
+
+func (setHandler *SetHandler) applyCpusetToInfraContainer(pod v1.Pod) error{
+	cpuset := setHandler.poolConfig.SelectPool(types.DefaultPoolID).CPUs
+	if cpuset.IsEmpty() {
+		//Nothing to set. We will leave the container running on the Kubernetes provisioned default cpuset
+		log.Println("WARNING: for some reason DEFAULT cpuset to set was quite empty. I left it untouched.")
+		return nil
+	}
+	containerID := pod.Status.ContainerStatuses[0].ContainerID
+	trimmedCid := strings.TrimPrefix(containerID, "docker://")
+	var pathToSearchContainer, pathToContainerCpusetFile, infraContainerID string
```

Comment:
this looks somewhat copy-pasted. you call this function already after we have identified the cpuset root directory
you could simply return it , or save it during the first iteration, and simply pass it onto this function